src/lua/LuaInterfaceGame.cpp

```diff
@@ -385,3 +388,18 @@ bool LuaInterfaceGame::EventCancelPactRequest(PactType pt, unsigned char cancele
         return onPactCancel.call<bool>(pt, canceledByPlayerId, targetPlayerId);
     return true; // always accept pact cancel if there is no handler
 }
+
+void LuaInterfaceGame::EventSuggestPact(const PactType pt, unsigned char suggestedByPlayerId, unsigned char targetPlayerId, const unsigned duration)
+{
+    kaguya::LuaRef onPactCancel = lua["onSuggestPact"];
+    if (onPactCancel.type() == LUA_TFUNCTION)
+    {
+        bool luaResult = onPactCancel.call<bool>(pt, suggestedByPlayerId, targetPlayerId, duration);
+        if (luaResult)
```

Comment:
Why not take both results? True = Accept, False = Cancel/Decline

Reply:
Done, aii now calls cancel pact if lua methods returns false.